### DISCUSSION TITLE
Cannot get fields.attachment from issue search

### DISCUSSION THREAD

**Discussion by alfredopalhares** [pycontribs/jira#1073] [Q&A]
Before raising any bug be sure that you used the latest release and that it still reproduces with the latest master. No other version is supported.

**Describe the bug**
I isse has no attachment Id filed, even tough on the web interface I can see them, 

**To Reproduce**
```
  issues_with_att = jira.search_issues(jql_str = 'PROJECT={proj} and attachments is not EMPTY'.format(proj = project_key))
    print(issues_with_att)
    for att in issues_with_att:
        print(att.key)
        print(att.fields.attachment
```
1. Any additional steps or considerations that happen before or after.

**Expected behavior**
I should be able to iterate thourgh the fileds.

**Stack Trace**
```
Traceback (most recent call last):
  File "/home/masterkorp/Documents/RenaltyXAi/repos/jira-migrator/./main.py", line 174, in <module>
    main()
  File "/usr/lib/python3.9/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.9/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.9/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/home//./main.py", line 64, in main
    print(att.fields.attachment)
AttributeError: 'PropertyHolder' object has no attribute 'attachment'
```

**Version Information**
Type of Jira instance:
- [x] Jira Cloud (Hosted by Atlassian)
- [x] Jira Server or Data Center (Self-hosted)
Python Interpreter: 3.9.5
jira-python: 3.0.1 and master
OS: Arch Linux
Other Dependencies:

**Additional context**
I can see and download all the attachements thorugh the web interface, same credentials.

**Comment by adehad** [comment]
In this case where 'attachment' is not preset, can specify what fields to get when using the `search_issues` function.

```python

    issues_with_att = jira.search_issues(
        jql_str = 'PROJECT={proj} and attachments is not EMPTY'.format(proj = project_key),
        fields=["attachment"]
    )
    print(issues_with_att)
    for att in issues_with_att:
        print(att.key)
        print(att.fields.attachment)

**Reply by adehad** [reply]
I've edited the comment as the `fields` argument replaces what fields are fetched rather than updating the existing list.

**Comment by laurikoobas** [comment]
I encountered the same issue and there is a solution:
https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-issue-search/#api-rest-api-3-search-get
> Note: All navigable fields are returned by default.

So you have to use `fields=['*all']` to get all the fields. Apparently attachments are considered non-navigable.
